app/assets/javascripts/discourse/app/components/modal/history.js

```diff
@@ -0,0 +1,337 @@
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import Category from "discourse/models/category";
+import Component from "@glimmer/component";
+import I18n from "I18n";
+import Post from "discourse/models/post";
+import { categoryBadgeHTML } from "discourse/helpers/category-link";
+import { iconHTML } from "discourse-common/lib/icon-library";
+import { sanitizeAsync } from "discourse/lib/text";
+import { inject as service } from "@ember/service";
+
+function customTagArray(val) {
+  if (!val) {
+    return [];
+  }
+  if (!Array.isArray(val)) {
+    val = [val];
+  }
+  return val;
+}
+
+export default class History extends Component {
+  @service dialog;
+  @service site;
+  @service currentUser;
+  @service siteSettings;
+
+  @tracked loading;
+  @tracked postRevision;
+  @tracked viewMode = this.site?.mobileView ? "inline" : "side_by_side";
+
+  constructor() {
+    super(...arguments);
+    this.refresh(this.args.model.postId, this.args.model.postVersion);
+  }
+
+  get loadFirstDisabled() {
+    return (
+      this.loading ||
+      this.postRevision?.current_revision <= this.postRevision?.first_revision
+    );
+  }
+
+  get loadPreviousDisabled() {
+    return (
+      this.loading ||
+      (!this.postRevision.previous_revision &&
+        this.postRevision.current_revision <=
+          this.postRevision.previous_revisionI)
+    );
+  }
+
+  get loadNextDisabled() {
+    return (
+      this.loading ||
+      this.postRevision?.current_revision >= this.postRevision?.next_revision
+    );
+  }
+
+  get loadLastDisabled() {
+    return (
+      this.loading ||
+      this.postRevision?.current_revision >= this.postRevision?.next_revision
+    );
+  }
+
+  get displayRevisions() {
+    return this.postRevision?.version_count > 2;
+  }
+
+  get modalTitleKey() {
+    return this.args.model.post.version > 100
+      ? "history_capped_revisions"
+      : "history";
+  }
+
+  get previousVersion() {
+    return this.postRevision?.current_version - 1;
```

Comment:
If `postRevision` is nullish, then this is going to return `NaN`. We should either remove the `?.`, or we improve the handling for this situation